glean-core/android/src/main/java/mozilla/telemetry/glean/scheduler/GleanLifecycleObserver.kt

```diff
@@ -0,0 +1,44 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+* License, v. 2.0. If a copy of the MPL was not distributed with this
+* file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.telemetry.glean.scheduler
+
+import androidx.lifecycle.Lifecycle
+import androidx.lifecycle.LifecycleObserver
+import androidx.lifecycle.OnLifecycleEvent
+import mozilla.telemetry.glean.Glean
+import mozilla.telemetry.glean.GleanMetrics.GleanBaseline
+
+/**
+ * Connects process lifecycle events from Android to Glean's handleEvent
+ * functionality (where the actual work of sending pings is done).
+ */
+internal class GleanLifecycleObserver : LifecycleObserver {
+    /**
+     * Calls the "background" event when entering the background.
+     */
+    @OnLifecycleEvent(Lifecycle.Event.ON_STOP)
+    fun onEnterBackground() {
+        // We're going to background, so store how much time we spent
+        // on foreground.
+        GleanBaseline.duration.stopAndSum(this)
```

Comment:
Having all this already in here means it will just start working once we implement what's currently stubbed out /o\ I think that's really cool